app/src/main/java/com/github/ravifrancesco/softwaredevproject/LineOfSight.java

```diff
@@ -0,0 +1,151 @@
+
+    }
+
+    /**
+     * This method computes a boolean that indicates if a single POIPoint is visible from
+     * the user's location.
+     *
+     * @param poiPoint      POIPoint to determine if it is visible.
+     * @param userIndexes   indexes representing the user's location on the elevation map grid.
+     * @param userLongitude longitude of the user's location (in degrees).
+     * @param userAltitude  altitude of the user's location (in meters).
+     * @return              <code>true</code> if the POIPoint is visible from the user's location.
+     * 	                    <code>false</code> otherwise.
+     */
+    private boolean isVisible(POIPoint poiPoint, Pair<Integer, Integer> userIndexes,
+                              double userLongitude, int userAltitude) {
+
+        Pair<Integer, Integer> poiIndexes = elevationMap
+                .getIndexesFromCoordinates(poiPoint.getLatitude(), poiPoint.getLongitude());
+        double poiLongitude = poiPoint.getLongitude();
+        int poiAltitude = (int) poiPoint.getAltitude();
+
+        double slope = (poiAltitude - userAltitude) / (poiLongitude - userLongitude);
+
+        List<Pair<Integer, Integer>> line = bresenham(  userIndexes.first, userIndexes.second,
+                                                        poiIndexes.first, poiIndexes.second);
+
+         return line.parallelStream()
+                .map(p -> elevationMap.getAltitudeAtLocation(p.first, p.second) -
+                        computeMaxElevation(userLongitude, userAltitude, p.second, slope) >
+                        ELEVATION_DIFFERENCE_THRESHOLD)
+                .reduce(true, (vis, v) -> vis &&  v);
+
+    }
+
+    /**
+     * This method draws the line between the user's location and the POIPoint on the
+     * elevation map grid. To do this the Bresenham algorithm is used:
+     *
+     * See <a href="Bresenham's line algorithm">https://en.wikipedia.org/wiki/Bresenham%27s_line_algorithm</a>
+     *
+     * @param x1    x index of the user's location on the elevation map grid.
+     * @param y1    y index of the user's location on the elevation map grid.
+     * @param x2    x index of the POIPoint on the elevation map grid.
+     * @param y2    y index of the POIPoint on the elevation map grid.
+     * @return      a List of indexes representing the line in the elevation map grid.
+     */
+    private List<Pair<Integer, Integer>> bresenham(int x1, int y1, int x2, int y2) {
+
+        int m_new = 2 * (y2 - y1);
+        int slope_error_new = m_new - (x2 - x1);
+
+        List<Pair<Integer, Integer>> line = new ArrayList<>();
+
+        for (int x = x1, y = y1; x <= x2; x++) {
+
+            line.add(new Pair<>(x, y));
+
+            // Add slope to increment angle formed
+            slope_error_new += m_new;
+
+            // Slope error reached limit, time to
+            // increment y and update slope error.
+            if (slope_error_new >= 0)
+            {
```

Comment:
Brackets should be the same everywhere
```suggestion
            if (slope_error_new >= 0){
```